tests/inverted-index-test.js

```diff
@@ -121,38 +124,119 @@ describe('Inverted index class creates index', () => {
+    invertedIndex.createIndex('validFile.json', validFile);
+    const index = invertedIndex.createIndex('anotherValidFile.json',
+      anotherValidFile);
+    it('looks up terms in the index of a file `fileName`', () => {
+      const searchResult = invertedIndex.searchIndex(index, 'validFile.json',
+       'the', ['system'], ['animals'])['validFile.json'];
+      expect(searchResult).toBeTruthy();
+      expect(searchResult.animals).toEqual([0, 1]);
+      expect(searchResult.the).toEqual([0, 1, 2]);
+      expect(searchResult.system).toEqual([1, 2]);
+    });
+
+    it('looks up terms in the index of all files if fileName is undefined',
+    () => {
+      const searchResult = invertedIndex.searchIndex(index, undefined,
+      ['and', ['fly']], ['the']);
+      const validFileResult = searchResult['validFile.json'];
+      const anotherValidFileResult = searchResult['anotherValidFile.json'];
+      expect(Object.keys(searchResult).length).toEqual(2);
+      expect(validFileResult.and).toEqual([1]);
+      expect(anotherValidFileResult.and).toEqual([1]);
+      expect(validFileResult.fly).toEqual([]);
+      expect(anotherValidFileResult.fly).toEqual([1, 2]);
+      expect(validFileResult.the).toEqual([0, 1, 2]);
+      expect(anotherValidFileResult.the).toEqual([1]);
+    });
+
+    it('looks up simultaneous terms in index of file(s)', () => {
+      const searchResult = invertedIndex.searchIndex(index, undefined,
+       ['of to'], ['the animals']);
+      const validFileResult = searchResult['validFile.json'];
+      const anotherValidFileResult = searchResult['anotherValidFile.json'];
+      expect(validFileResult['of to']).toEqual([0]);
+      expect(anotherValidFileResult['of to']).toEqual([]);
+      expect(validFileResult['the animals']).toEqual([0, 1]);
+      expect(anotherValidFileResult['the animals']).toEqual([]);
+    });
+
+    it('returns empty array for term if term is not in the index', () => {
+      const searchResult = invertedIndex.searchIndex(index,
+      'anotherValidFile.json', 'beauty', ['in', 'our'],
+       'eyes')['anotherValidFile.json'];
+      expect(searchResult.beauty).toEqual([]);
+      expect(searchResult.in).toEqual([]);
+      expect(searchResult.our).toEqual([]);
+      expect(searchResult.eyes).toEqual([]);
+    });
+
+    it('returns a message if no term is given', () => {
+      const searchResult = invertedIndex.searchIndex(index,
+      'anotherValidFile.json');
+      expect(searchResult).toEqual('Terms cannot be empty');
+    });
+
+    it('returns a message if index is not an object', () => {
+      const searchResult = invertedIndex.searchIndex([],
+      'anotherValidFile.json');
+      expect(searchResult).toEqual('Invalid index Object');
+    });
+
+    it('returns a message if index is an empty object', () => {
+      const searchResult = invertedIndex.searchIndex({},
+      'anotherValidFile.json');
+      expect(searchResult).toEqual('Invalid index Object');
     });
   });
 });
```

Comment:
Good job Impressive test coverage.  But can't seem to find search for words having other characters attached to them. Example: 'what#$@'